src/Engine/SearchEngine.php

```diff
@@ -25,21 +27,105 @@ public function __construct(
 
     public function execute(?string $query, array $parameters = []): SearchResult
     {
-        $page = max(1, (int) ($parameters['p'] ?? 1));
-        $sort = (string) ($parameters['sort'] ?? '');
-
         $metadata = $this->metadataFactory->getMetadataFor($this->index->document);
+        $indexUid = $this->indexNameResolver->resolve($this->index);
```

Comment:
```suggestion
        $indexName = $this->indexNameResolver->resolve($this->index);
```

I have used the term `indexName` in all other places.